apisix/upstream.lua

```diff
@@ -293,7 +293,7 @@ function _M.set_by_route(route, api_ctx)
     end
 
     if nodes_count > 1 then
-        local checker = fetch_healthchecker(up_conf)
+        local checker = fetch_healthchecker(up_conf, api_ctx.var.host)
```

Comment:
```suggestion
        local checker = fetch_healthchecker(up_conf, api_ctx.var.upstream_host)
```
Not the one sent by client

Reply:
@spacewander ok, will update